rosettascio/rsciio/image/api.py

```diff
@@ -80,9 +79,13 @@ def file_writer(filename, signal, scalebar=False, scalebar_kwds=None,
         scalebar_kwds = dict()
     scalebar_kwds.setdefault('box_alpha', 0.75)
     scalebar_kwds.setdefault('location', 'lower left')
-
-    if rgb_tools.is_rgbx(data):
-        data = rgb_tools.rgbx2regular_array(data)
+    try:
+        # HyperSpy uses struct arrays to store RGBA data
+        from hyperspy.misc import rgb_tools
```

Comment:
Yes. Probably this is a bad idea. I did it to avoid passing somehow the information that it is RGB data, but on a second though, that's maybe a better approach.

Also, I've just realized that phenom and tiff still depend on hyperspy's rgb_tools. I will uncheck this task from the progress tracker above.

Reply:
> Also, I've just realized that phenom and tiff still depend on hyperspy's rgb_tools. I will uncheck this task from the progress tracker above.

I notice this too but I came to the conclusion that if the data follows hyperspy structured numpy array for RGB(A), which is quite specific, it most likely come from hyperspy! As this can't be sure this is correct, this may need to change at some point and this can be done later.

It seems that this is used in various place in hyperspy, shall we could move it to the the public API of rosettasciio and used it in hyperspy?

Reply:
Sounds good. HyperSpy stores RGB data in struct arrays to enable seamless multi-dimensional RGB datasets. Without moving `rgb_tools` to RosettaSciIO it will be difficult for other packages to read/save RGB data in hspy/zspy format.